### PR TITLE
Be resilient to getting back no spans to tag.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/20961

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/Core/ReferenceHighlighting/ReferenceHighlightingViewTaggerProvider.cs [hunk 1]
```diff
@@ -69,6 +69,8 @@ protected override ITaggerEventSource CreateEventSource(ITextView textView, ITex
 
         protected override IEnumerable<SnapshotSpan> GetSpansToTag(ITextView textViewOpt, ITextBuffer subjectBuffer)
         {
+            // Note: this may return no snapshot spans.  We have to be resilient to that
+            // when processing the TaggerContext<>.SpansToTag below.
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/ReferenceHighlighting/ReferenceHighlightingViewTaggerProvider.cs]: Can we test this with a text buffer of some unsupported content type?

Comment (dpoeschl) [inline src/EditorFeatures/Core/ReferenceHighlighting/ReferenceHighlightingViewTaggerProvider.cs]: Or an integration test to cover the actual scenario if we understand it (Dave's repro looks too complicated).

Comment (dpoeschl) [inline src/EditorFeatures/Core/ReferenceHighlighting/ReferenceHighlightingViewTaggerProvider.cs]: Do other taggers not have this problem (haven't looked)?

Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/ReferenceHighlighting/ReferenceHighlightingViewTaggerProvider.cs]: I spot checked them, and they're ok.

**Pull request merged**